### DISCUSSION TITLE
[Deployment] Unsuccessful following the documentation

### DISCUSSION THREAD

**Discussion by padupe** [Oriflame/backstage-plugins#151] [Q&A] [answered]
I am studying the implementation of the plugin in my company.
Some questions arose:

1. Is deployment only possible via Azure?
2. Is it necessary to configure scores for entities linked to the Catalog individually?

These doubts arose after following the project's README faithfully and not obtaining satisfactory results.

**Comment by jvilimek** [comment, marked as answer]
Hi @padupe and thanks for the questions.

> 1. Is deployment only possible via Azure?

It's one way, how to implement the process (SharePoint form -> azure storage). This was the way of Oriflame. But you may have your own storage (e.g. github repo), see https://github.com/Oriflame/backstage-plugins/issues/111 or you can implement your own API to retrieve the data. Maybe it would be good to mention this in the readme to bring more clarity to this.

> 2. Is it necessary to configure scores for entities linked to the Catalog individually?

Not sure, what you mean. Can you please clarify what you want to achieve and why?

**Reply by padupe** [reply]
Hello @jvilimek!
I thank you for your willingness to respond and help with the implementation here...

The solution to the first question proves to have potential.
I will validate its usage using GitHub App for authentication.

As for the second:
- Does each project need to have an individual scoring criterion?

For example:
Project A
- Documentation
  - Has 100
  - Partial 50
  - Does not have 0
- Tests
  - Unitary and Integration 100
  - Only one "type" 50
  - Does not have 0

Project B
- Documentation
  - Has 100
  - Partial 50
  - Does not have 0
- Tests
  - Unitary and Integration 100
  - Only one "type" 50
  - Does not have 0

**Reply by jvilimek** [reply]
Thanks for clarification. I believe the scales as described by you exists only in the [template] (https://github.com/Oriflame/backstage-plugins/blob/main/plugins/score-card/sample-data/%40template.json) from where these are used to populate the SharePoint form (example of how to implement the review process).

**Comment by padupe** [comment]
@jvilimek I used #111's indication, replicating the entire structure presented in [sample-data](https://github.com/Oriflame/backstage-plugins/tree/main/plugins/score-card/sample-data) and got the following error:
![image](https://user-images.githubusercontent.com/72894267/208983372-9df09204-b98b-4efc-a758-9536940806a0.png)

My `app-config.yaml`
```yaml
[...]

proxy:
  [...]
  '/scorecards':
    target: 'https://raw.githubusercontent.com/raizen-devops/tech-hub/feat/8002-score-card/raizen/'
    headers:
      Authorization: 'token ${GITHUB_PATOKEN}'

[...]

scorecards:
  jsonDataUrl: /api/proxy/scorecards/

[...]
```

Structure of remote files:
![image](https://user-images.githubusercontent.com/72894267/208984531-35caa38e-b839-4a55-9836-61e9af6713c7.png)

**Reply by jvilimek** [reply]
Thanks for trying. There are few issues I am afraid. Right now there is a breaking change ongoing in the way, how the path to the data is created. It was `{jsonDataUrl}/{name}.json` and in the new, not yet released package, the path is `{jsonDataUrl}/{namespace}/{kind}/{name}.json`.. let's try to use some older version of the sample data, would you be so kind?
Another question is related to the way, how GitHub is handling http404/403/..errors. what does the network tab in developers tools say?

**Reply by padupe** [reply]
I don't know if I understood correctly. Do you suggest that I use an older version of the plugin?

**Reply by jvilimek** [reply]
I am sorry it took so long. Have you already fixed the issue?
I was suggesting using the older version of the sample data like e.g. [all.json (commit 07b0e7cf35632f8a6bb6ecef1806045dc90eaa32)](https://raw.githubusercontent.com/Oriflame/backstage-plugins/07b0e7cf35632f8a6bb6ecef1806045dc90eaa32/plugins/score-card/sample-data/all.json).
But now there is updated version 0.6.0 of the plugin that should be aligned with the current sample data. Will you try to use the updated version [0.6.0](https://www.npmjs.com/package/@oriflame/backstage-plugin-score-card/v/0.6.0)?
